pkg/cedar/translated_client.go

```diff
@@ -50,3 +59,133 @@ func (c TranslatedClient) FetchSystems(logger *zap.Logger) (models.SystemShorts,
 	}
 	return systems, nil
 }
+
+type cedarIntake map[string]bool
+
+func requiredFields() cedarIntake {
+	return cedarIntake{
+		"ID":                      true,
+		"EUAUserID":               true,
+		"Status":                  true,
+		"Requester":               true,
+		"Component":               true,
+		"BusinessOwner":           true,
+		"BusinessOwnerComponent":  true,
+		"ProductManager":          true,
+		"ProductManagerComponent": true,
+		"ISSO":                    false,
+		"TRBCollaborator":         false,
+		"OITSecurityCollaborator": false,
+		"EACollaborator":          false,
+		"ProjectName":             true,
+		"ExistingFunding":         true,
+		"FundingNumber":           false,
+		"FundingSource":           false,
+		"BusinessNeed":            true,
+		"Solution":                true,
+		"EASupportRequest":        true,
+		"ProcessStatus":           true,
+		"ExistingContract":        true,
+		"UpdatedAt":               true,
+		"SubmittedAt":             true,
+	}
+}
+
+func validateSystemIntakeForCedar(intake *models.SystemIntake, logger *zap.Logger) error {
+	var res []string
+	v := reflect.ValueOf(*intake)
+	requiredFields := requiredFields()
+	for i := 0; i < v.NumField(); i++ {
```

Comment:
I'm not happy with the way I did this, but I couldn't think of a better way to do this. Please feel free to suggest ideas here.